### PR TITLE
Do not hangup in `onConnectionStateChanged()`

### PR DESCRIPTION
**Pull request by SimonBrandner**

### CODE REVIEW

File: pkg/peer/webrtc_callbacks.go [hunk 1]
```diff
@@ -76,8 +75,6 @@ func (p *Peer[ID]) onConnectionStateChanged(state webrtc.PeerConnectionState) {
 	p.logger.Infof("Connection state changed: %v", state)
 
 	switch state {
-	case webrtc.PeerConnectionStateFailed, webrtc.PeerConnectionStateDisconnected, webrtc.PeerConnectionStateClosed:
-		p.sink.Send(LeftTheCall{event.CallHangupUserHangup})
```
Comment (dbkr) [inline pkg/peer/webrtc_callbacks.go:80]: We should still do it in failed (and presumably closed) so I'd just remove `webrtc.PeerConnectionStateDisconnected` from the middle.

**Review by dbkr — APPROVED** [review]
Thanks, lgtm!

**Comment by daniel-abramov** [comment]
Sorry @SimonBrandner, I'm missing a bit of a context. Why don't we want to hang up on a disconnect? (I could assume that it's because of the reconnect that would be expected on the client side?)

**Comment by SimonBrandner** [comment]
> Sorry @SimonBrandner, I'm missing a bit of a context. Why don't we want to hang up on a disconnect? (I could assume that it's because of the reconnect that would be expected on the client side?)

In the js-sdk we never send a hangup event in the case where `connectionstatechange` is fired, so we're assuming that we fire it incorrectly here. From trying to understand the documentation we gathered that we might be able to recover from `disconnected` and we've removed it here.

Further context: https://matrix.to/#/!dXHEZfLxIQzAELxzQd:matrix.org/$JJmHD9GTUZjmVLERRHZhDpoSXPMXqgcAeiVzpYKzB0Q?via=matrix.org&via=element.io&via=one.ems.host (and messages below)

**Comment by daniel-abramov** [comment]
Makes sense, thanks!

**Pull request merged**